### PR TITLE
add random state generate in DataLoader worker

### PR DESCRIPTION
**Pull request by heavengate**
### PR types
Bug fixes

### PR changes
APIs

### Describe
**add random state generate in DataLoader worker to generate different seed for each worker**
numpy random state remains same in `fork` sub-process, which may cause performance loss in some model, such as https://tanelp.github.io/posts/a-bug-that-plagues-thousands-of-open-source-ml-projects/

### CODE REVIEW

File: python/paddle/fluid/dataloader/state.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import numpy as np
+
+__all__ = ['_generate_states']
```
Comment (chenwhql) [inline python/paddle/fluid/dataloader/state.py]: 好像非公开API层级 all 列表都要置空？麻烦确认一下，参考PR：https://github.com/PaddlePaddle/Paddle/pull/31912

File: python/paddle/fluid/dataloader/state.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+# The function `_generate_states` is adapted from `numpy.random.SeedSequence`
+# from https://github.com/numpy/numpy/blob/main/numpy/random/bit_generator.pyx
+# It's MIT licensed, here is the copyright:
+
+# Copyright (c) 2015 Melissa E. O'Neill
```
Comment (chenwhql) [inline python/paddle/fluid/dataloader/state.py]: 有这个copy right，上面的paddle copyright还需要吗？

File: python/paddle/fluid/dataloader/state.py (hunk 1 above)
Comment (heavengate) [reply python/paddle/fluid/dataloader/state.py]: Done，thanks!

File: python/paddle/fluid/dataloader/state.py (hunk 2 above)
Comment (heavengate) [reply python/paddle/fluid/dataloader/state.py]: ci里有个copyright_checker会自动添加这个，删除了`state.py`文件，函数放到`worker.py`里了，这个函数也只在`_worker_loop`函数里用到

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**